services/authentication-service/src/main/java/ru/kharevich/authenticationservice/controller/ex/ExternalUserErrorDecoder.java

```diff
@@ -0,0 +1,51 @@
+package ru.kharevich.authenticationservice.controller.ex;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import feign.Response;
+import feign.Util;
+import feign.codec.ErrorDecoder;
+import jakarta.persistence.EntityNotFoundException;
+import jakarta.xml.bind.ValidationException;
+import lombok.extern.slf4j.Slf4j;
+import ru.kharevich.authenticationservice.exceptions.ExternalValidationException;
+import ru.kharevich.authenticationservice.exceptions.RepeatedDataException;
+
+import java.io.IOException;
+import java.util.Map;
+
+import static ru.kharevich.authenticationservice.utils.constants.AuthenticationServiceResponseConstants.UNKNOWN_EXTERNAL_EXCEPTION_CAUSE;
+
+@Slf4j
+public class ExternalUserErrorDecoder implements ErrorDecoder {
+    @Override
+    public Exception decode(String methodKey, Response response) {
+        String causeMessage = extractExceptionMessage(response);
+
+        switch(response.status()) {
+            case 400:
+                return new ExternalValidationException(causeMessage);
+            case 409:
+                return new RepeatedDataException(causeMessage);
+        }
+        return new RuntimeException();
+    }
+
+    private final String extractExceptionMessage(Response response){
```

Comment:
Не следует забывать про применение автоформатирования.
```suggestion
    private final String extractExceptionMessage(Response response) {
```

Почему метод объявлен с модификатором final?